examples/apps/api-gateway-multiple-origin-cors/index.js

```diff
@@ -0,0 +1,57 @@
+// Demonstrates a Lambda function that provides CORS headers for many origins.
+
+const cors = require("./cors-util");
+
+// define a whitelist of allowed origins
+const allowedOrigins = [
+    "http://127.0.0.1",
+    "https://*.example.com",
+    "https://*.amazon.com"
+].map(cors.compileURLWildcards);
+
+/**
+ * Demonstrates a simple endpoint that accepts GET requests.
+ * 
+ * In most cases, browsers do not perform CORS preflight requests when using
+ * the GET method, so we do not have to handle OPTIONS requests.
+ * We include all CORS headers in the GET response.
+ */
+exports.handleRoot = async (event, context) => {
+    const origin = event.headers.Origin || event.headers.origin;
+    const allowedMethods = ["GET"];
+
+    // return an empty response, with CORS headers
+    return {
+        headers: Object.assign(cors.makeHeaders(origin, allowedOrigins, allowedMethods), {
+            // use Object.assign to include additional headers
+        }),
+        statusCode: 204
+    };
+};
+
+/**
+ * Demonstrates an endpoint that accepts DELETE requests.
+ * 
+ * In this case, the browser will perform a CORS preflight request, so we must
+ * handle OPTIONS requests and provide the CORS headers.
+ * When the browser makes the DELETE request, we only need to provide the origin.
+ */
+exports.handleTest = async (event, context) => {
+    const origin = event.headers.Origin || event.headers.origin;
+    const allowedMethods = ["OPTIONS", "DELETE"];
+
+    if (event.httpMethod === "OPTIONS") {
+        // return an empty response, with all CORS headers
+        return {
+            headers: cors.makeHeaders(origin, allowedOrigins, allowedMethods),
```

Comment:
A higher level of abstraction seems reasonable. I'm not sure if there would be a case where a user might want to add extra headers to the response. It might be sufficient to add a parameter for `Access-Control-Max-Age`, which seems to be the only other header relevant to a preflight response.

Reply:
I've gone ahead and done that